src/Data/CartItem/Cart.ts

```diff
@@ -48,14 +52,9 @@ export function numberOfCartItems(): number | null {
 export async function checkoutCart(): Promise<DataItem> {
   const cartItems: DataItem[] = await load(() => CartItem.all().take())
 
-  const products: ProductInstance[] = []
-  for (const item of cartItems) {
-    const productId = item.get('productId')
-    if (typeof productId !== 'string') continue
-
-    const product = await load(() => Product.get(productId))
-    if (product) products.push(product)
-  }
+  const products: DataItem[] = cartItems.map(
+    (item) => item.get('product') as DataItem,
```

Comment:
* [ ] FYI the validity of the cast depends on multiple conditions spread over the codebase.
  * How can I be sure product is a reference attribute?
  * How can I be sure the reference is not null?
  
   Maybe we should use a type guard.